File: 장아연/week4/10828.py
# https://www.acmicpc.net/problem/10828

import sys
N = int(sys.stdin.readline())
st = []

for _ in range(N):
    arr = sys.stdin.readline().split()
    if (arr[0] == "push"):
        # print(int(arr[1]))
        st.append(int(arr[1]))
    elif (arr[0] == "pop"):
        if(st):
            print(st.pop())
        else:
            print(-1)
    elif (arr[0] == "size"):
        print(len(st))
    elif (arr[0] == "top"):
        if (st):
            print(st[-1])
        else:
            print(-1)
    elif (arr[0] == "empty"):
        if(st):
            print(0)
        else:
            print(1)

File: 김현아/10828.cpp
//220409
//백준 10828번 스택 (실버 4)
//https://www.acmicpc.net/problem/10828

#include <iostream>
#include <string>
#include <stack>

using namespace std;
stack<int> s;

int main() {
	int N;
	string order;
	cin >> N;
	while (N--) {
		cin >> order;
		if (order == "push") {
			int num;
			cin >> num;
			s.push(num);
		}
		else if (order == "pop") {
			if (s.empty()) {
				cout << -1 << "\n";
			}
			else {
				cout << s.top();
				s.pop();
			}
		}
		else if (order == "size") {
			cout << s.size();
		}
		else if (order == "empty") {
			if (s.empty()) {
				cout << 1;
			}
			else {
				cout << 0;
			}
		}
		else if (order == "top") {
			if (s.empty()) {
				cout << -1 << "\n";
			}
			else {
				cout << s.top();
			}
		}
	}
	return 0;
}